packages/react-server/core/renderMiddleware.js

```diff
@@ -590,18 +592,27 @@ function renderScripts(pageObject, res) {
 }
 
 function renderStylesheets (pageObject, res) {
-	pageObject.getHeadStylesheets().forEach((styleSheet) => {
-		if (styleSheet.href) {
-			res.write(`<link rel="stylesheet" type="${styleSheet.type}" media="${styleSheet.media}" href="${styleSheet.href}" ${PAGE_CSS_NODE_ID}>`);
-		} else if (styleSheet.text) {
-			res.write(`<style type="${styleSheet.type}" media="${styleSheet.media}" ${PAGE_CSS_NODE_ID}>${styleSheet.text}</style>`);
-		} else {
-			throw new Error("Style cannot be rendered because it has neither an href nor a text attribute: " + styleSheet);
-		}
+
+	const styles = PageUtil.standardizeStyles(pageObject.getHeadStylesheets());
+
+	const stylePromises = styles.map(styleP => {
+		return styleP.then(styleSheet => {
+			if (!styleSheet) {
+				// skip. a promise resolving to nothing is the only way to decide
+				// to not output a stylesheet if you return a promise
+				return;
+			}
+			if (styleSheet.href) {
+				res.write(`<link rel="stylesheet" type="${styleSheet.type}" media="${styleSheet.media}" href="${styleSheet.href}" ${PAGE_CSS_NODE_ID}>`);
+			} else if (styleSheet.text) {
+				res.write(`<style type="${styleSheet.type}" media="${styleSheet.media}" ${PAGE_CSS_NODE_ID}>${styleSheet.text}</style>`);
+			} else {
+				throw new Error("Style cannot be rendered because it has neither an href nor a text attribute: " + styleSheet);
+			}
+		});
 	});
 
-	// resolve immediately.
-	return Q("");
+	return Q.all(stylePromises);
```

Comment:
`Q.all` maintains backwards-compatibility in the case that a malformed style object was passed, but it introduces new behavior where failure to figure out what stylesheet to load can cause the page to fail to render. Thoughts?